Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -4346,431 +4346,305 @@ void CPhysicalGeometry::PrepareAdjacency(CConfig *config) {
-    if (prism_flip > 0) cout << "There has been a re-orientation of the PRISM volume elements." << endl;
-    if (hexa_flip > 0) cout << "There has been a re-orientation of the HEXAHEDRON volume elements." << endl;
-    if (pyram_flip > 0) cout << "There has been a re-orientation of the PYRAMID volume elements." << endl;
+    if (tria_flip+quad_flip+tet_flip+hexa_flip+pyram_flip+prism_flip+
+        quad_error+pyram_error+prism_error+hexa_error == 0) {
+      cout << "All volume elements are correctly orientend." << endl;
+    }
   }
 
 }
 
-void CPhysicalGeometry::Check_BoundElem_Orientation(CConfig *config) {
-
-  unsigned long Point_1_Surface, Point_2_Surface, Point_3_Surface, Point_4_Surface,
-  iElem_Domain, Point_Domain = 0, Point_Surface, iElem_Surface,
-  line_flip = 0, triangle_flip = 0, quad_flip = 0;
-  su2double test_1, test_2, test_3, test_4, *Coord_1, *Coord_2, *Coord_3, *Coord_4,
-  *Coord_5, a[3] = {0.0,0.0,0.0}, b[3] = {0.0,0.0,0.0}, c[3] = {0.0,0.0,0.0}, n[3] = {0.0,0.0,0.0}, test;
-  unsigned short iDim, iMarker, iNode_Domain, iNode_Surface;
-  bool find;
-
-  for (iMarker = 0; iMarker < nMarker; iMarker++) {
-
-    if (config->GetMarker_All_KindBC(iMarker) != INTERNAL_BOUNDARY) {
+void CPhysicalGeometry::Check_BoundElem_Orientation(const CConfig *config) {
 
-      for (iElem_Surface = 0; iElem_Surface < nElem_Bound[iMarker]; iElem_Surface++) {
+  unsigned long line_flip = 0, tria_flip = 0, quad_flip = 0, quad_error = 0;
 
-        iElem_Domain = bound[iMarker][iElem_Surface]->GetDomainElement();
-        for (iNode_Domain = 0; iNode_Domain < elem[iElem_Domain]->GetnNodes(); iNode_Domain++) {
-          Point_Domain = elem[iElem_Domain]->GetNode(iNode_Domain);
-          find = false;
-          for (iNode_Surface = 0; iNode_Surface < bound[iMarker][iElem_Surface]->GetnNodes(); iNode_Surface++) {
-            Point_Surface = bound[iMarker][iElem_Surface]->GetNode(iNode_Surface);
-            if (Point_Surface == Point_Domain) {find = true; break;}
-          }
-          if (!find) break;
-        }
+  SU2_OMP_PARALLEL_(reduction(+:line_flip,tria_flip,quad_flip,quad_error)) {
 
-        /*--- 2D grid, line case ---*/
+  /*--- Lambda to test tetrahedrons. ---*/
+  auto checkTetra = [this](unsigned long Point_1, unsigned long Point_2,
+                           unsigned long Point_3, unsigned long Point_4) {
+    const auto Coord_1 = nodes->GetCoord(Point_1);
+    const auto Coord_2 = nodes->GetCoord(Point_2);
+    const auto Coord_3 = nodes->GetCoord(Point_3);
+    const auto Coord_4 = nodes->GetCoord(Point_4);
+    constexpr int nDim = 3;
+    su2double a[nDim]={0.0}, b[nDim]={0.0}, c[nDim]={0.0}, n[nDim]={0.0};
+    GeometryToolbox::Distance(nDim, Coord_2, Coord_1, a);
+    GeometryToolbox::Distance(nDim, Coord_3, Coord_1, b);
+    GeometryToolbox::Distance(nDim, Coord_4, Coord_1, c);
+    GeometryToolbox::CrossProduct(a,b,n);
+    return GeometryToolbox::DotProduct(nDim,n,c) < 0.0;
```

Comment:
These functions are similar to the ones used to check the volume elements, I did not make this another member function of CGeometry or a free function of the geometry toolbox mostly out of laziness, but I guess some arguments in favor are to avoid crowding the CGeometry namespace, and because here what I wanted to "automate" was the boiler plate of fetching coordinates.
In any case I don't mind if someone moves this in the future.